govcd/vapp.go

```diff
@@ -786,6 +798,328 @@ func (vapp *VApp) AddIsolatedNetwork(newIsolatedNetworkSettings *VappNetworkSett
+		fenceMode = types.FenceModeNAT
+	}
+
+	networkConfigurations := vapp.VApp.NetworkConfigSection.NetworkConfig
+	var services types.NetworkFeatures
+	var natService types.NatService
+	if newNetworkSettings.NatEnabled != nil {
+		natService = types.NatService{IsEnabled: *newNetworkSettings.NatEnabled, NatType: "ipTranslation", Policy: "allowTrafficIn"}
+	}
+	if newNetworkSettings.FirewallEnabled != nil {
+		services = types.NetworkFeatures{
+			FirewallService: &types.FirewallService{IsEnabled: *newNetworkSettings.FirewallEnabled},
+			NatService:      &natService,
+		}
+	}
+	vappConfiguration := types.VAppNetworkConfiguration{
+		NetworkName: orgNetwork.Name,
+		Configuration: &types.NetworkConfiguration{
+			FenceMode: fenceMode,
+			Features:  &services,
+			ParentNetwork: &types.Reference{
+				HREF: orgNetwork.HREF,
+			},
+			RetainNetInfoAcrossDeployments: newNetworkSettings.RetainIpMacEnabled,
+		},
+		IsDeployed: false,
+	}
+	networkConfigurations = append(networkConfigurations,
+		vappConfiguration)
+
+	return updateNetworkConfigurations(vapp, networkConfigurations)
+
+}
+
+// UpdateNetwork updates vApp networks (isolated or connected to Org VDC network)
+// Returns pointer to types.NetworkConfigSection or error
+func (vapp *VApp) UpdateNetwork(newNetworkSettings *VappNetworkSettings, orgNetwork *types.OrgVDCNetwork) (*types.NetworkConfigSection, error) {
+	task, err := vapp.UpdateNetworkAsync(newNetworkSettings, orgNetwork)
+	if err != nil {
+		return nil, err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("%s", combinedTaskErrorMessage(task.Task, err))
+	}
+
+	vAppNetworkConfig, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return nil, fmt.Errorf("error getting vApp networks: %#v", err)
+	}
+
+	return vAppNetworkConfig, nil
+}
+
+// UpdateNetworkAsync asyncronously updates vApp networks (isolated or connected to Org VDC network).
+// Returns task or error
+func (vapp *VApp) UpdateNetworkAsync(networkSettingsToUpdate *VappNetworkSettings, orgNetwork *types.OrgVDCNetwork) (Task, error) {
+	util.Logger.Printf("[TRACE] UpdateNetworkAsync with values: %#v and connect to org network: %#v", networkSettingsToUpdate, orgNetwork)
+	currentNetworkConfiguration, err := vapp.GetNetworkConfig()
+	if err != nil {
+		return Task{}, err
+	}
+	var networkToUpdate types.VAppNetworkConfiguration
+	var networkToUpdateIndex int
+	for index, networkConfig := range currentNetworkConfiguration.NetworkConfig {
+		uuid, err := GetUuidFromHref(networkConfig.Link.HREF)
+		if err != nil {
+			return Task{}, err
+		}
+
+		if uuid == networkSettingsToUpdate.Id {
+			networkToUpdate = networkConfig
+			networkToUpdateIndex = index
+			break
+		}
+	}
+
+	if networkToUpdate == (types.VAppNetworkConfiguration{}) {
+		return Task{}, fmt.Errorf("not found network to update with Id %s", networkSettingsToUpdate.Id)
+	}
+
+	if networkToUpdate.Configuration.Features.NatService != nil && networkSettingsToUpdate.NatEnabled != nil {
```

Comment:
Would it be a big risk/breaking change to use `*bool` and assign directly instead of converting each value?

Reply:
It used by other functions - which will be impacted, needs to be verified, etc.

Reply:
Alright. Lets live with it.